platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md

```diff
@@ -77,76 +74,84 @@ VFS teams are required to use Pact for contract testing in the following scenari
-4. The provider (`vets-api`) runs a rake task to verify the pacts. This verification task replays the requests defined in the pacts against the provider (`vets-api`) and validates that the actual response matches the expected response.
-5. Results of the verification are published back to the broker.
+The process is a collaborative effort where front-end (FE) and back-end (BE) engineers should communicate and may need to iterate on the pacts to get them successfully verified.
 
-Builds depend on successful verification results to deploy.
+#### Front-end workflow
 
-*Note: In future iterations of the VSP Pact Process, we plan to integrate the rake verification task into CircleCI as part of the build step.* 
+1. FE engineers write Pact tests in feature branches in vets-website.
+2. The vets-website CI pipeline runs the contract tests job.
+   - The job runs all Pact tests, which generate pacts if they pass.
+   - The pacts are then published to a central broker.
+3. The vets-website CI pipeline runs a job to check `can-i-deploy`.
+   - The `can-i-deploy` task returns the verification statuses of all pacts against vets-api's master branch.
+     This determines whether it's safe to merge this vets-website feature branch.
+   - If there is no status yet for a pact, it will poll for 30 seconds while the pacts are asynchronously verified.
+   - If the status of any pact can't be determined, the job times out with a failure.
+4. The broker triggers the verification job in the vets-api CI pipeline, which publishes its results to the broker.
+5. If all pacts in the branch are successfully verified, the branch can be merged.
+6. If verification has failed for any pact, FE engineers should discuss with BE engineers to resolve the failure.
+   **The branch should not be merged until the `can-i-deploy` task is successful.**
+   FE should adjust the pact or BE should update provider states or API responses to accommodate the pact.
+   If the resolution only involves BE changes, wait for those changes to get merged and re-run the vets-website CI pipeline.
 
+#### Back-end workflow
 
-![](https://i.imgur.com/zQMyDS0.png)
+1. BE engineers update API endpoints and/or provider states in a feature branch in vets-api.
+2. The vets-api CI pipeline runs the verification job, which verifies the all of the latest pacts on the vets-website master branch.
+3. If the verification job passes, the branch can be merged.
+4. If the verification has failed for any pact, BE engineers should discuss with FE engineers to resolve the failure.
+   BE should adjust provider states or API responses to accommodate the pact or FE should update the pact.
+   If the resolution only involves FE changes, wait for those changes to get merged and re-run the vets-api CI pipeline.
 
-### Implementation details
+![](https://i.imgur.com/zQMyDS0.png)
```

Comment:
Do we like having this large image here? I'm wondering if it's helpful

Reply:
I suppose we could do better, but the diagram I made a while back is a bit outdated now. This generic diagram is what we've got for now, so I'll leave it alone.